### PR TITLE
QA Kubernetes Tasks

### PR DESCRIPTION
**Pull request by joshmeek**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
QA'd the k8s tasks and found that the API secret was required, adjusted so it can be provided as None. This required tests to be updates slightly. Also added tests for each task's in/out of cluster configuration connections.

## Why is this PR important?
Secret was required and some connection options went untested

### CODE REVIEW

File: src/prefect/tasks/kubernetes/deployment.py [hunk 1]
```diff
@@ -87,7 +88,9 @@ def run(
                 "A dictionary representing a ExtensionsV1beta1Deployment must be provided."
             )
 
-        kubernetes_api_key = Secret(kubernetes_api_key_secret).get()
+        kubernetes_api_key = None
+        if kubernetes_api_key_secret:
+            kubernetes_api_key = Secret(kubernetes_api_key_secret).get()
```
Comment (cicdw) [inline src/prefect/tasks/kubernetes/deployment.py]: So the way I read this, by _default_ the key is still required, this just allows you to explicitly provide `None` if you want to bypass it?  I assume needing the key is still the more common use case?

Comment (joshmeek) [reply src/prefect/tasks/kubernetes/deployment.py]: Yeah the way I believe this should be used is through the API key and remote connection

Comment (joshmeek) [reply src/prefect/tasks/kubernetes/deployment.py]: But you can override it by explicitly providing None

Comment (cicdw) [reply src/prefect/tasks/kubernetes/deployment.py]: Ok cool that makes sense

**Comment by joshmeek** [comment]
Yeah I’ll update the docstrings, good call

**Pull request merged**